include/meltpooldg/heat/laser.hpp

```diff
@@ -40,6 +46,8 @@ namespace MeltPoolDG::Heat
      */
     double laser_intensity;
 
+    LaserImpactType impact_type;
```

Comment:
`const`?

Reply:
This is not const because of this statement in the ctor:
```
    if (laser_data.impact_type == "volumetric")
      impact_type = volumetric;
    else if (laser_data.impact_type == "interface")
      impact_type = interface;
    else
      AssertThrow(false, ExcMessage("Unknown laser impact type! Abort..."));
```

Do you have a suggestion make this work with `const LaserImpactType impact_type; `?

Reply:
You could try:
```suggestion
    impact_type( laser_data.impact_type == "volumetric" ? volumetric : interface )
```
The assert can be omitted since it is already caught within the parameter reader.

Reply:
I think I'll go with @peterrum 's solution because it can include the Assert if `laser_data.impact_type` is set to a string that does not fit the pattern.